File: dubai_shipping_tool/frontend/src/types/task.ts
export type TaskStatus = 'PENDING' | 'RUNNING' | 'WAITING_LOGIN' | 'SUCCESS' | 'FAILED'

export interface TaskInfo {
  task_id: string
  task_type: string
  status: TaskStatus
  message: string
  started_at: number | null
  finished_at: number | null
  error_detail: string | null
}

export interface OrderRow {
  order_nr: string
  partner_sku: string
  target_shipped_at: string | null
}

export interface OrdersPreview {
  total: number
  page: number
  page_size: number
  data: OrderRow[]
}

File: dubai_shipping_tool/backend/app/schemas/task.py
"""Shared API response models."""
from __future__ import annotations

from typing import Optional

from pydantic import BaseModel


class TaskCreateResponse(BaseModel):
    task_id: str
    task_type: str
    status: str
    message: str


class TaskInfoResponse(BaseModel):
    task_id: str
    task_type: str
    status: str
    message: str
    started_at: Optional[float] = None
    finished_at: Optional[float] = None
    error_detail: Optional[str] = None


class OrderRow(BaseModel):
    order_nr: str
    partner_sku: str
    target_shipped_at: Optional[str] = None


class InventoryRow(BaseModel):
    SKU: str
    当前库存: int


class InTransitRow(BaseModel):
    SKU: str
    在途库存: int
